include/nana/gui/widgets/skeletons/textbase.hpp

```diff
@@ -266,6 +265,7 @@ namespace skeletons
 					}
 
 					if (bytes)
+						const char * le_boms[] = { "\xEF\xBB\xBF", "\xFF\xFE", "\xFF\xFE\x0\x0" }; //BOM for little-endian
```

Comment:
I'm afraid this modification will change original flow of the code since there aren't curly brances after `if` statement.
So it should be like this.
```cpp
				if (bytes) {
						const char * le_boms[] = { "\xEF\xBB\xBF", "\xFF\xFE", "\xFF\xFE\x0\x0" }; //BOM for little-endian
						ofs.write(le_boms[static_cast<int>(encoding)], bytes);
				}
```

And I'm not really sure this subtle change would actually gives us better optimized code.
Have you actually checked assembly code and measured performance difference before and after?

Reply:
@beru,
Thank you for reviewing the code, any localization of variables simplifies the work of the compiler. The increase is small, of course, but these edits do not particularly spoil the style of the code.

Reply:
hmm.. did you compile this?
`le_boms `is not defined at the moment it is "used" in the next line outside the if were it is defined.